libsolidity/ast/AST.h

```diff
@@ -715,6 +726,10 @@ class FunctionDefinition: public CallableDeclaration, public Documented, public
 	{
 		return Declaration::isVisibleInContract() && isOrdinary();
 	}
+	bool isVisibleViaContractName() const override
+	{
+		return visibility() >= Visibility::Public;
```

Comment:
not internal?

Reply:
We could actually allow internal ones... but they would be empty objects with no use... they couldn't be called and don't have a signature and they can't be assigned to anything...

Reply:
Isn't it possible to call them internally?

Reply:
```
contract A {
    function f() internal virtual pure returns (uint) {return 2; }
}
contract B is A {
    function f() internal virtual override pure returns (uint) {return 3; }
}
contract C is B {
    function g() public pure returns (uint) {
        return A.f();
    }
}
```

Reply:
That's ``isVisibleInDerivedContract`` not this one... but yeah, I'm not sure how to be clearer in the distinction - I renamed it to ``isVisibleViaContractTypeAccess``, maybe that helps, but arguably that's still also what happens in the inheritance case... it's only meant to return ``true``, in case something can be accessed via the contract type name in *any* context, not merely when if inheriting.

Reply:
Ah ok, then I see!